components/right-sidebar/SocialMediaModal.tsx

```diff
@@ -0,0 +1,144 @@
+"use client";
+
+import { Loader2, X } from "lucide-react";
+import {
+  SubmitHandler,
+  useForm,
+  Controller,
+  FieldError,
+} from "react-hook-form";
+import toast from "react-hot-toast";
+import { zodResolver } from "@hookform/resolvers/zod";
+
+import { updateUser } from "@/lib/actions/user.actions";
+import {
+  ISocialMediaSchema,
+  IUserSchema,
+  SocialMediaSchema,
+} from "@/lib/validations/UserSchema";
+import { Social, User } from "@prisma/client";
+import Input from "../shared/ui/Input";
+import { socialMediaIconList } from "@/lib/constants/socialMediaList";
+import Button from "../shared/ui/Button";
+
+const SocialMediaModal = ({
+  user,
+  onClose,
+}: {
+  user: User & { socialMedia?: Social[] };
+  onClose: () => void;
+}) => {
+  const useFormHelpers = useForm<ISocialMediaSchema>({
+    defaultValues: {
+      socialMedia: user.socialMedia ?? [],
+    },
+    resolver: zodResolver(SocialMediaSchema),
+  });
+
```

Comment:
A nice feature to add would be some keyboard shortcut to our modal:

```js
  useEffect(() => {
    const handleEsc = (event: KeyboardEvent) => {
      if (event.key === "Escape") {
        onClose();
      }
    };

    window.addEventListener("keydown", handleEsc);

    return () => {
      window.removeEventListener("keydown", handleEsc);
    };
  }, []);
```

This will run when the component mounts and start listening for the escape key. It will then run your "close" function when the escape key is pressed.

```js
    return () => {
      window.removeEventListener("keydown", handleEsc);
    };
```

This section just removes the event listener when the component unmounts so that we don't try to and call "onClose" when the component isn't even available.

This is a small thing but makes for a nice user experience.